src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,211 @@
+ * context).
+ */
+public final class ObserverEqArray extends ObjectContract {
+
+  /** The run-time result of calling the observer: an array of literals. */
+  private Object value;
+
+  /**
+   * The maximum delta between the expected and actual for which both numbers (doubles or floats)
+   * are still considered equal.
+   */
+  private static final double FLOATING_POINT_DELTA = 1e-15;
+
+  /**
+   * Creates a new ObserverEqArray.
+   *
+   * @param value the run-time result of calling the observer: an array of literals
+   * @param isAccessible the accessibility predicate
+   */
+  public ObserverEqArray(Object value, AccessibilityPredicate isAccessible) {
+    this.value = value;
+    if (!isLiteralType(value.getClass().getComponentType(), isAccessible)) {
+      throw new RandoopBug(
+          String.format(
+              "Cannot represent %s as a literal",
+              StringsPlume.toStringAndClass(value.getClass().getComponentType())));
+    }
+  }
+
+  /**
+   * Returns true if values of the class can be represented as literals in Java source code.
+   *
+   * @param cls a class
+   * @param isAccessible the accessibility predicate
+   * @return true iff the class is a primitive, boxed primitive, String, Class, or accessible Enum
+   */
+  public static boolean isLiteralType(Class<?> cls, AccessibilityPredicate isAccessible) {
+    if (cls == Class.class || cls == String.class) {
+      return true;
+    }
+    if (cls.isEnum() && isAccessible.isAccessible(cls)) {
+      return true;
+    }
+    return PrimitiveTypes.isBoxedPrimitive(cls) || cls.isPrimitive();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof ObserverEqArray)) {
+      return false;
+    }
+    ObserverEqArray other = (ObserverEqArray) o;
+    return Arrays.equals(toObjectArray(value), toObjectArray(other.value));
+  }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(toObjectArray(value));
+  }
+
+  @Override
+  public String toCodeString() {
+    if (value.getClass().getComponentType() == float.class
+        || value.getClass().getComponentType() == double.class) {
+      return String.format(
+          "org.junit.Assert.assertArrayEquals(x0, %s, %s);",
+          newArrayExpression(), FLOATING_POINT_DELTA);
+    } else {
+      return String.format("org.junit.Assert.assertArrayEquals(x0, %s);", newArrayExpression());
+    }
+  }
+
+  /**
+   * Returns a Java array constructor expression for this, e.g., "new int[] {1,2,3}".
+   *
+   * @return a Java array constructor expression for this
+   */
+  private String newArrayExpression() {
+    return "new " + value.getClass().getCanonicalName() + arrayComponentsToString();
+  }
+
+  /**
+   * Converts an Object into its array representation.
+   *
+   * @param value the given Object
+   * @return its array representation
+   */
+  private Object[] toObjectArray(Object value) {
```

Comment:
I don't think this method is necessary.  Overloads of the methods that you are calling should statically dispatch to the correct implementation, without creating any new arrays.